modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/MicrosoftAzureAttachTranscriptionOperationHandler.java

```diff
@@ -93,21 +97,41 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
     // Check which tags/flavors have been configured
     ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(
             workflowInstance, Configuration.none, Configuration.none, Configuration.many, Configuration.one);
-    List<String> targetTagOption = tagsAndFlavors.getTargetTags();
-    List<MediaPackageElementFlavor> targetFlavorOption = tagsAndFlavors.getTargetFlavors();
+    List<String> targetTags = tagsAndFlavors.getTargetTags();
     // Target flavor is mandatory
     MediaPackageElementFlavor targetFlavor = tagsAndFlavors.getSingleTargetFlavor();
+    String language = StringUtils.trimToNull(operation.getConfiguration(OPT_LANGUAGE));
 
     try {
       // Get transcription file from the service
-      MediaPackageElement original = service.getGeneratedTranscription(mediaPackage.getIdentifier().toString(), jobId);
-      MediaPackageElement transcription = original;
+      MediaPackageElement transcription = service.getGeneratedTranscription(mediaPackage.getIdentifier().toString(),
+              jobId);
+
+      // Get return values from the service
+      try {
+        Map<String, Object> returnValues = service.getReturnValues(mediaPackage.getIdentifier().toString(), jobId);
+        autoDetectedLanguage = (String) returnValues.get("autoDetectedLanguage");
+      } catch (NullPointerException e) {
+        logger.warn("Key missing in return values", e);
+      } catch (TranscriptionServiceException e) {
+        logger.warn("Something went wrong when trying to receive return values", e);
+      }
 
       // Set the target flavor
+      if (language != null) {
```

Comment:
Afaik `parseFlavor` only cares about `/` and just ignores other special characters.

Personally, I woudn't expect the `+` to dissappear. The whole replacing `___` with something else already feels pretty hacky and not great to work with from a workflow creator perspective, implicitly removing other characters would probably cause them even more headache?

But maybe there is a reason a flavor cannot end in a `+` (or other special charactrs?) that I'm forgetting?

Reply:
AFAIK there's no reason it can't.  I guess my approach would be that seeing a flavour of `foo/` would mean that something is wrong and there's a subflavour missing, so seeing `foo/bar+` would be equally an error case.

I suspect you're right in that it won't *actually* break anything though, and I agree that the current approach isn't great - although I don't have a better suggestion.  Let's just leave this as is.